packages/eas-cli/src/commands/workflow/runs.ts

```diff
@@ -0,0 +1,151 @@
+import { WorkflowRunFragment, WorkflowRunStatus } from '../../graphql/generated';
+import { AppQuery } from '../../graphql/queries/AppQuery';
+import { WorkflowRunQuery } from '../../graphql/queries/WorkflowRunQuery';
+import Log from '../../log';
+import formatFields from '../../utils/formatFields';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export type WorkflowRunResult = {
+  id: string | null;
+  status: string | null;
+  gitCommitMessage?: string | null;
+  gitCommitHash?: string | null;
+  startedAt: string | null;
+  finishedAt: string | null;
+  workflowId: string | null;
+  workflowName: string | null;
+  workflowFileName: string | null;
+};
+
+function processWorkflowRuns(
+  runs: WorkflowRunFragment[],
+  params: {
+    workflowFileName?: string | undefined;
+    status?: string | undefined;
+  }
+): WorkflowRunResult[] {
+  const { workflowFileName, status } = params;
+  return runs
+    .filter(run => {
+      if (workflowFileName && run.workflow?.fileName !== workflowFileName) {
+        return false;
+      }
+      if (status && run.status !== status) {
+        return false;
+      }
+      return true;
+    })
+    .map(run => {
+      const finishedAt = run.status === WorkflowRunStatus.InProgress ? null : run.updatedAt;
+      return {
+        id: run.id ?? null,
+        status: run.status ?? null,
+        gitCommitMessage: run.gitCommitMessage ?? null,
+        gitCommitHash: run.gitCommitHash ?? null,
+        startedAt: run.createdAt ?? null,
+        finishedAt,
+        workflowId: run.workflow?.id ?? null,
+        workflowName: run.workflow?.name ?? null,
+        workflowFileName: run.workflow?.fileName ?? null,
+      };
+    });
+}
+
+export default class WorkflowRunList extends EasCommand {
+  static override description =
+    'list recent workflow runs for this project, with their IDs, statuses, and timestamps';
+
+  static override flags = {
+    workflow: Flags.string({
+      description:
+        'If present, the query will only return runs for the specified workflow file name',
+      required: false,
+    }),
+    status: Flags.enum({
+      description: 'If present, filter the returned runs to select those with the specified status',
+      required: false,
+      options: Object.values(WorkflowRunStatus),
+    }),
+    ...EasJsonOnlyFlag,
+    limit: getLimitFlagWithCustomValues({ defaultTo: 10, limit: 100 }),
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectId,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags } = await this.parse(WorkflowRunList);
+    const {
+      projectId,
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(WorkflowRunList, {
+      nonInteractive: true,
+    });
+
+    const workflowFileName = flags.workflow;
+    const status = flags.status;
+    const limit = flags.limit ?? 10;
+
+    let runs: WorkflowRunFragment[] = [];
```

Comment:
```suggestion
    let runs: WorkflowRunFragment[];
```
(so typescript catches fallthrough cases, which we don't want to allow)